src/App.js

```diff
@@ -1,26 +1,60 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import contacts from './contacts.json';
+import Button from './components/Button'
+import Table from './components/Table';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+export default class App extends React.Component {
+  state = {
+    contacts: contacts.slice(0, 5)
+  };
+
+  addRandomContact = () => {
+    const randomNumber = Math.floor(
+      Math.random() * (contacts.length - this.state.contacts.length) +
+        this.state.contacts.length
+    );
+
+    this.setState({
+      contacts: [
+        ...this.state.contacts.filter((f) => f !== contacts[randomNumber]),
+        contacts[randomNumber],
+      ],
+    });
+  };
+  sortName = () => {
+    const contactSortName = this.state.contacts.sort((a, b) => {
+      if (a.name > b.name) return 1;
+      if (a.name < b.name) return -1;
+      return 0;
```

Comment:
Aquí esto también te valdría y así vas probando cosas nuevas....

const contactSortName = this.state. contacts.sort((a, b) => a.name.localeCompare(b.name))